### ISSUE TITLE
Verify result always returning "verify-no-match"

### ISSUE THREAD

**Issue by ov1d1u** [ars3niy/fprint_vfs5011#9]
Hi,

Thank you for this driver, I got my fingerprint scanner on a Lenovo M5400 working on Linux. Unfortunately I cannot get the sensor to validate my fingerprint, everytime I get "Verify result: verify-no-match (done)" when I run "fprintd-verify" (of course, I enrolled my fingertip before doing this).

fprint_demo shows a distorted image (http://i.imgur.com/eltf3k4.png) every time I do a fingertip scanning, so I'm asking if not this is the issue.

Anyway, thanks for your work!

**Comment by Eugeny** [comment]
For me, increasing ``MAX_OFFSET`` has removed fingerprint image distortions but I still couldn't achieve reliable verification.

**Comment by Vince-vd** [comment]
Have you guys ever found a good solution for this? I have the same problem with my lenovo t440s and the fingerprint reader is probably the only thing I really miss since switching to ubuntu a few months ago.

Eugeny what did you change the MAX_OFFSET to? Because I don't really see any difference when playing around with the values.

**Comment by Eugeny** [comment]
@Vince-vd I tried values up to 1000, and it improved the image, but I can't say that it had any effect on the recognition quality

**Comment by Vince-vd** [comment]
Ok thank you @Eugeny , I'll just keep typing my password then I guess :)

**Comment by EAnushan** [comment]
I can confirm that this issue is still present. The image is distorted and I can only match every 1/25 attempts.

**Comment by elder** [comment]
The same problem on Dell Vostro 5470. I can't authenticate using my fingerprint reader.

**Comment by ptonelli** [comment]
same issue for lenovo x1 carbon gen3

**Comment by marcocamma** [comment]
Hi same observation as ptonelli (on the same machine, running debian sid)

**Comment by Nestorfish** [comment]
With my T550, I noticed the distortion is created by vfs5011_rescale_image(). As I understand it, it tries to compensate for finger swipe speed variation by doing a correlation between the left and right sections of the capture.
Replacing the function with a simple data copy avoids the distortion and allows slow swipe.
@ars3niy Do you have the datasheet or other information to understand what data is effectively present on the "right side" of the capture ?

**Comment by ercoppa** [comment]
@ars3niy logs with distortions: https://www.dropbox.com/s/1udq3hxg1rmncm5/vfs5011debug.tar.gz
@Nestorfish can you please make fork with your changes?

**Comment by Nestorfish** [comment]
@ercoppa Done. As said in the commit message, other tests must be done to try to get better results.

**Comment by ercoppa** [comment]
Since the driver has been included upstream (https://bugs.freedesktop.org/show_bug.cgi?id=61692), maybe we should move our discussion there. I agree with you that without a datasheet is hard to understand what it is the right fix that needs to be implemented. Right now, at least for me, the driver is not reliable.

**Comment by IvanFrederiks** [comment]
@ars3niy 
Looks like this issue was fixed in upstream repository. fprint-enroll, fprint-verify and fprint_demo work fine with latest ppa build of libfprint (1:0.6.0-git20151216-1-0ppa1~trusty1). Tested on ThinkPad Edge E540.

P.S.
Fingerprint GUI still crashes during enroll after first swipe.

**Comment by andyboeh** [comment]
You're right, thanks for reporting it here! It works fine on my Arch Linux with libfprint-git. It was actually fixed by this commit: https://cgit.freedesktop.org/libfprint/libfprint/commit/?id=6e8d5cd6a199f83fb5f862a8421f8582608032f5 so it's going to take some time until it reaches stable.

**Comment by dr3mro** [comment]
I am facing same issue now on Thinkpad x250 with arch linux

**Comment by CheesecakeCG** [comment]
@dr3mro Yeah, same here on the T450S, I find it really surprising that the sensitivity options are not exposed in a config file.

**Comment by Mdubois12** [comment]
I am also experiencing this issue on a Thinkpad T450s, anyone have a solution?

**Comment by Toxblh** [comment]
Maybe too late. But main knowledge here - https://github.com/iafilatov/libfprint#common-problems
You should swipe your finger instead of touch. fprintd used "long" image for that.
So as write - guys also provide "touch" version the driver

**Comment by andirsun** [comment]
> Maybe too late. But main knowledge here - https://github.com/iafilatov/libfprint#common-problems
> You should swipe your finger instead of touch. fprintd used "long" image for that.
> So as write - guys also provide "touch" version the driver

How can I install the touch version of the driver ?

**Comment by WieeRd** [comment]
I'm 5 years late but I solved this by installing the [`libfprint-tod-git`](https://aur.archlinux.org/packages/libfprint-tod-git) AUR package on Arch.
It builds on top of this [`tod`](https://aur.archlinux.org/packages/libfprint-tod-git) branch/fork on GitLab. Not sure about convenient ways to install this on other distros though.
Hopefully one day this 11 years old issue gets resolved on the main branch of libfprint.

Thank you so much @Toxblh for leaving that comment because otherwise I would've been completely clueless.
